duolingo.py

```diff
@@ -28,13 +29,21 @@ class DuolingoException(Exception):
 class Duolingo(object):
     USER_AGENT = "Python Duolingo API/{}".format(__version__)
 
-    def __init__(self, username, password):
+    def __init__(self, username, password, *, jwt=None, session_file=None):
```

Comment:
password might now be optional, since it's not always required when either session_file or jwt is provided

Reply:
It still can't be optional. If the JWT expires or isn't valid, the code goes straight to trying the password to generate a new one.

Reply:
I don't think it's necessary to expose both password and JWT at the same time. Because in most cases if JWT is provided it would work. You can just throw an exception that JWT isn't valid. In the very beginning can add a check that one of `password`, `jwt` or `session_file` must  be provided.
So, for example, in my script I temporarily store JWT in ENV of the bash session and I don't need to type of store my password anywhere.

Reply:
Yeah, I suppose that is an option, to just fail if jwt is invalid and no password is provided.

Reply:
I've just checked, this JWT is issued for 1 year on login, so maybe it's better if user double checks what's wrong with provided JWT instead of forced login in attempt.

Reply:
Yeah, that's fair. Now changed so that password, jwt, or session_file is required, rather than just password